web/app/js/components/TapQueryForm.jsx

```diff
@@ -112,11 +100,12 @@ class TapQueryForm extends React.Component {
       let resourcesByNs = props.resourcesByNs;
       let authoritiesByNs = props.authoritiesByNs;
       let namespaces = Object.keys(resourcesByNs).sort();
-      let resourceNames  = getResourceList(resourcesByNs, state.query.namespace);
-      let toResourceNames = getResourceList(resourcesByNs, state.query.toNamespace);
-      let authorities = getResourceList(authoritiesByNs, state.query.namespace);
+      let resourceNames  = getResourceList(resourcesByNs, props.query.namespace);
```

Comment:
I saw that we were maintaining `query` values into props and state and, having in mind that we need to update the parent state, I thought that avoiding unnecessary renders to update both across the component was a good improvement :)

Most of the changes on this file are regarding to avoid duplicity in state and props `query`.

Reply:
Hi @cynthia-sg, thanks for the explanation. I agree that it's a good change to avoid unnecessary renders, great 👁! 

Because these changes are not directly related to updating React, I would prefer to make those changes in a separate PR. That way, this PR will be solely focused on updating React and the fewest lines of code possible are changed. We'll be able to review it and merge it in more quickly, and there is less chance of introducing errors.

After it's been merged, if you could create a new issue for other improvements to the component, explaining what the changes were and the benefits, that would be great 👍